fj_2024_lesson_5

mirpribili opened the pull request.

mirpribili committed:
init

mirpribili committed:
init

mirpribili committed:
fix

mirpribili committed:
fix

mirpribili committed:
first Aspect

mirpribili committed:
add time to Aspect

mirpribili committed:
add Aspect to class controller

mirpribili committed:
add Aspect to class controller

mirpribili committed:
add Aspect to classes

mirpribili committed:
add Locations

mirpribili committed:
add test

mirpribili committed:
add logging-starter

lParzivall commented on app/src/main/java/ru/tbank/App.java:
```diff
@@ -1,12 +1,20 @@
 
     public static void main(String[] args) {
-        System.out.println("Hi");
+        logger.info("Запуск приложения...");
```
не лучшее решение все это в main засовывать

https://www.baeldung.com/running-setup-logic-on-startup-in-spring

мне нравится больше всего ApplicationRunner

lParzivall commented on app/src/main/java/ru/tbank/aspect/LoggingAspect.java:
```diff
@@ -0,0 +1,29 @@
+public class LoggingAspect {
+    private static final Logger logger = LoggerFactory.getLogger(LoggingAspect.class);
+
+    //@Around("(execution(* ru.tbank.controller..*(..)) && @target(ru.tbank.annotation.LogControllerExecution)) || @annotation(ru.tbank.annotation.LogMainExecution)")
```
уберем везде лишние коммы

lParzivall commented on app/src/main/java/ru/tbank/controller/CategoryController.java:
```diff
@@ -0,0 +1,63 @@
+@LogControllerExecution
+public class CategoryController {
+    private final CategoryRepository repository;
+    private static final Logger logger = LoggerFactory.getLogger(DataInitializer.class);
```
можно через анноташку, сейчас логер никто так не получает, не модно

lParzivall commented on app/src/main/java/ru/tbank/controller/CategoryController.java:
```diff
@@ -0,0 +1,63 @@
+    private final CategoryRepository repository;
+    private static final Logger logger = LoggerFactory.getLogger(DataInitializer.class);
+
+    public CategoryController(CategoryRepository repository) {
```
тут тоже ломбок бы справился

lParzivall commented on app/src/main/java/ru/tbank/controller/CategoryController.java:
```diff
@@ -0,0 +1,63 @@
+    public CategoryController(CategoryRepository repository) {
+        this.repository = repository;
+    }
+    @LogControllerExecution
```
зачем, если есть над классом?

lParzivall commented on app/src/main/java/ru/tbank/controller/CategoryController.java:
```diff
@@ -0,0 +1,63 @@
+    public ResponseEntity<Category> getCategoryById(@PathVariable int id) {
+        logger.info("Получение категории с ID: {}", id);
+        Category category = repository.findById(id);
+        if (category == null) {
```
так делать не очень хорошо, лишний код в контроллере
бест практис будет написать exceptionHandler @controllerAdvice

в сервисе используй Optional.ofNullable.orElseThrow и как раз в хендлере будешь это исключение ловить и отдавать нужный ResponseEntity.notFound

lParzivall commented on app/src/main/java/ru/tbank/model/Category.java:
```diff
@@ -0,0 +1,14 @@
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
```
вот, ты же умеешь в ломбок)

lParzivall commented on app/src/main/java/ru/tbank/repository/CategoryRepository.java:
```diff
@@ -0,0 +1,41 @@
+public class CategoryRepository {
+    private static final Logger logger = LoggerFactory.getLogger(DataInitializer.class);
+    private final ConcurrentHashMap<Integer, Category> categories = new ConcurrentHashMap<>();
+    private int currentId = 0;
```
этого не просили, но что тут с многопоточкой?)

lParzivall commented on app/src/main/java/ru/tbank/service/DataInitializer.java:
```diff
@@ -0,0 +1,69 @@
+
+import java.util.Arrays;
+
+@Service
```
тут Configuration больше подходит

lParzivall commented on app/src/main/java/ru/tbank/service/DataInitializer.java:
```diff
@@ -0,0 +1,69 @@
+    public DataInitializer(CategoryRepository categoryRepository, LocationRepository locationRepository) {
+        this.categoryRepository = categoryRepository;
+        this.locationRepository = locationRepository; // Инициализируем репозиторий городов
+        this.restTemplate = new RestTemplate();
```
тоже можно через бин вернуть, вдруг потом в сервисе понадобится RestTemplate ну и используем DI

lParzivall commented on app/src/main/java/ru/tbank/service/DataInitializer.java:
```diff
@@ -0,0 +1,69 @@
+    private final LocationRepository locationRepository; // Добавляем репозиторий городов
+    private final RestTemplate restTemplate;
+
+    public DataInitializer(CategoryRepository categoryRepository, LocationRepository locationRepository) {
```
ломбок сюда можно спокойно

lParzivall commented on app/src/main/java/ru/tbank/service/DataInitializer.java:
```diff
@@ -0,0 +1,69 @@
+
+        try {
+            // Загрузка категорий
+            String categoriesUrl = "https://kudago.com/public-api/v1.4/place-categories";
```
вынести в проперти

lParzivall commented on app/src/main/java/ru/tbank/service/DataInitializer.java:
```diff
@@ -0,0 +1,69 @@
+            String categoriesUrl = "https://kudago.com/public-api/v1.4/place-categories";
+            try {
+                Category[] categories = restTemplate.getForObject(categoriesUrl, Category[].class);
+                if (categories != null) {
```
Optional

mirpribili committed:
1. add @Logging
    private Logger logger;
2. del some commits
3. DataInitializer implements ApplicationRunner

mirpribili committed:
1. change @Logging to @Slf4j

mirpribili committed:
1. del not used imports

mirpribili committed:
add exceptionHandler @controllerAdvice

mirpribili committed:
add exceptionHandler @controllerAdvice

mirpribili committed:
add exceptionHandler @controllerAdvice

mirpribili committed:
fix private int->Integer currentId = 0;

mirpribili committed:
fix not used

lParzivall commented on app/src/main/java/ru/tbank/annotation/LogMainExecution.java:
```diff
@@ -0,0 +1,11 @@
+
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface LogMainExecution {
```
ты же вынес в стартер, отсюда нужно убрать
